server/ts/api/api_level.ts

```diff
@@ -526,6 +539,35 @@ export const initLevelApi = () => {
 
 		res.end();
 	});
+
+	// Fetches the specified leaderboards for the level
+	app.get('/api/level/:levelId/leaderboards/:leaderboardId', async (req, res) => {
+		let levelId = await verifyLevelId(req, res);
+		if (levelId === null) return;
+
+		let doc = await db.missions.findOne({ _id: levelId }) as MissionDoc;
+		let mission = Mission.fromDoc(doc);
+
+		let leaderboardId = req.params.leaderboardId;
+		let lbQueryInfo = Util.chooseDataByDatablockCompatibility(config.leaderboardSources, mission.datablockCompatibility);
+		let query = lbQueryInfo.find(x => x.id === leaderboardId);
+		if (query === undefined) {
+			res.status(400).send("400\nLevel does not contain the specified leaderboard.");
+			return null;
+		}
+
+		try {
+			let resp = await fetch(query.queryUrl.replace('{id}', levelId.toString()));
+			let json = await resp.json();
+
+			let respParsed = lbEndpointResponseSchema.parse(json);
+			respParsed.scores.sort((a, b) => a.placement - b.placement); // Sort the scores in place anyway, make sure we always return sorted
+			res.send(respParsed); // Just send it as it is, it's validated
+		} catch (e) {
+			res.status(400).send("400\nEndpoint schema did not return in the expected format.");
```

Comment:
I think this should return `502 bad gateway` here

Reply:
Sure